### PR TITLE
fixed array types in parsing routines + added special case for char[]

### PR DESCRIPTION
**Pull request by bluec0re**
Interpret char[] as char* in the pf format (use z instead of c).

### CODE REVIEW

**Comment by bluec0re** [comment]
Travis CI fails are because of https://github.com/radare/radare2-regressions/commit/d25032b064a99ee337dd0391c2514cdef5e7e279#diff-f4df7f39caee3d915656b6e2bcb33410

and unrelated to the changes.

File: shlr/tcc/tccgen.c [hunk 1]
```diff
@@ -661,10 +661,14 @@ static void type_to_str(char *buf, int buf_size,
         goto no_var;
     case VT_PTR:
         s = type->ref;
-        pstrcpy(buf1, sizeof(buf1), "*");
-        if (varstr)
-            pstrcat(buf1, sizeof(buf1), varstr);
-        type_to_str(buf, buf_size, &s->type, buf1);
+        if (t & VT_ARRAY) {
```
Comment (radare) [inline shlr/tcc/tccgen.c]: use tabs, not spaces

File: shlr/tcc/tccgen.c [hunk 2]
```diff
@@ -661,10 +661,14 @@ static void type_to_str(char *buf, int buf_size,
...
+            type_to_str(buf, buf_size, &s->type, NULL);
```
Comment (radare) [inline shlr/tcc/tccgen.c]: missing space before every (

File: shlr/tcc/tccgen.c [hunk 3]
```diff
@@ -661,10 +661,14 @@ static void type_to_str(char *buf, int buf_size,
...
+        } else {
+            pstrcpy(buf1, sizeof(buf1), "*");
+            if (varstr)
```
Comment (radare) [inline shlr/tcc/tccgen.c]: use braces even for 1 line conditionals

File: shlr/tcc/tccgen.c (hunk 1 above)
Comment (radare) [inline shlr/tcc/tccgen.c]: oh wait its ok, following tcc standards

**Pull request merged**